src/ORM/Marshaller.php

```diff
@@ -211,6 +211,14 @@ public function one(array $data, array $options = [])
             $entity->set($properties);
         }
 
+        // Don't flag clean association entities as
+        // dirty so we don't persist empty records.
+        foreach ($properties as $field => $value) {
+            if ($value instanceof EntityInterface) {
+                $entity->dirty($field, $value->dirty());
```

Comment:
This is where I think the big change is... a good one, though.

Does it make more sense to only traverse the properties that we know are from associations and that are allowed in the associations whitelist?